lib/purchases_flutter.dart

```diff
@@ -260,6 +274,9 @@ class Purchases {
         .toList();
   }
 
+  /// It is now recommended to use [Purchases.purchaseStoreProduct]
+  /// to make a purchase with a [StoreProduct] if you can.
```

Comment:
Is there a reason why we haven't just deprecated this function? It requires a `type` which is a source of confusion for a lot of devs that want to offer consumables. A lot of them forget to pass it and the purchase doesn't work 😢

Reply:
@vegaro No reason that I know of 😬  But I would be happy to deprecate this function, TBH 😇  It just looks super error prone to me